frontend/src/ui/D3Visualizations/Treemap.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useLayoutEffect, useRef } from 'react';
+import * as d3 from 'd3';
+import { TreeHierarchy } from '../../domain/hierarchy';
+
+export type TreemapProps = {
+  height: number;
+  width: number;
+  data: TreeHierarchy;
+};
+
+export default function Treemap({ height, width, data }: TreemapProps) {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  useLayoutEffect(() => {
+    renderTreemap();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  function renderTreemap() {
+    const container: HTMLDivElement | null = containerRef.current;
+
+    // create svg
+    const svg: d3.Selection<
+      SVGSVGElement,
+      unknown,
+      null,
+      undefined
+    > = d3
+      .select(container)
+      .append('svg')
+      .attr('width', width)
+      .attr('height', height);
+
+    // create hierarchical layout with data
+    const root: d3.HierarchyNode<TreeHierarchy> = d3
+      .hierarchy(data)
+      .sum((d: any) => d.transitionRate)
+      .sort((a: any, b: any) => b.transitionRate - a.transitionRate);
+
+    // initialize treemap
+    const treemap = d3.treemap().size([width, height]).padding(1);
+
+    treemap(root);
+
+    // select the nodes and set x, y position
+    const nodes = svg
+      .selectAll('g')
+      .data(root.leaves())
+      .enter()
+      .append('g')
+      .attr('transform', (d: any) => `translate(${d.x0},${d.y0})`);
+
+    // lighten colors
+    const fader = (color: string): string =>
+      d3.interpolateRgb(color, '#fff')(0.3);
+
+    // select color within d3.schemeSet3 (contains array of 12 colors)
+    const colorScale = d3.scaleOrdinal(d3.schemeSet3.map(fader));
```

Comment:
From the docs it seems like scaleOrdinal is for mapping discrete values to discrete values, but here we're using it to map transitionRate (continuous numbers) to colors (discrete strings). It seems to work though, each box gets a different color. Another option would be to use scaleQuantize or scaleQuantile which map continuous values to discrete values